build/index.js

```diff
@@ -228,13 +228,29 @@ async function buildDocumentFromURL(url) {
 async function buildLiveSamplePageFromURL(url) {
   const [documentURL, sampleID] = url.split("/_samples_/");
   const document = Document.findByURL(documentURL);
-  const liveSamplePage = kumascript.buildLiveSamplePage(
-    document.url,
-    document.metadata.title,
-    (await kumascript.render(document.url))[0],
-    { id: sampleID, createFlaw() {} }
-  );
-  return liveSamplePage.html;
+  if (!document) {
+    return `unable to find ${documentURL}`;
+  }
+  // Convert the lower-case sampleID we extract from the incoming URL into
+  // the actual sampleID object with the properly-cased live-sample ID.
+  for (const sampleIDObject of kumascript.getLiveSampleIDs(
```

Comment:
I'm wondering if we can keep the simplicity of the previous approach by changing the middleware @peterbe brought back in to also hold on to the `originalURL` [before lowercasing it](https://github.com/Gregoor/yari/blob/fc31de0f6ad7ead5bbf3a3573c4be3152903a1c1/server/middlewares.js#L19). Express allows you to attach arbitrary data to the request object, which is what a lot of middlewares do. We'd then have to make sure to pass it into this function. And it'd restrict the building to only work with correctly cased URLs, which yeah. I don't know if we want that. I guess it's a trade-off between caller simplicity vs implementation simplicity.

Reply:
That's a good thing to explore. I would prefer to keep the simplicity of the previous approach (by adding `originalURL` to the request object as you said), but it seems wrong to depend upon the proper casing in URL's (as you mentioned as well).